src/NuGet.Services.AzureSearch/Auxiliary2AzureSearch/Auxiliary2AzureSearchCommand.cs

```diff
@@ -90,6 +90,9 @@ public async Task ExecuteAsync()
                 _logger.LogInformation("Fetching new download count data from blob storage.");
                 var newData = await _auxiliaryFileClient.LoadDownloadDataAsync();
 
+                _logger.LogInformation("Removing invalid IDs and versions from the old data.");
+                CleanDownloadData(oldResult.Result);
```

Comment:
how does the invalid data make it into these files?

Reply:
It comes from stats DB.
https://github.com/NuGet/Engineering/issues/2638